Decorator/TextDecorator/EmojiMessageDecorator.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+namespace Decorator
+{
+    public class EmojiDecorator : MessageDecorator
+    {
+        public EmojiDecorator(IMessage message) : base(message)
+        {
+            string text = message.GetMessage();
+
+            Console.OutputEncoding = System.Text.Encoding.Unicode;
+            if (text.Contains(":-)"))
+                text = text.Replace(":-)", "\x263A");
+
+            if (text.Contains(":-("))
+                text = text.Replace(":-(", "\x2639");
+
+            if (text.Contains("<3"))
+                text = text.Replace("<3", "\x2764");
+
+            message.SetMessage(text);
```

Comment:
You didn't decorate the output. You changed the state of the original class.

Reply:
done